### PR TITLE
VACMS-12309 add document entity fields

### LINKED ISSUE #12309: Add custom VA specific fields to documentation entity
The content_model_documentation entity will have some pre-existing default fields that are useful for all projects.
However there are some fields that will only be needed for VA so should not be part of the contrib 
Example:  the airtable has a column to track which fields we push to lighthouse.

## AC's
- [x] Audit airtable spectool for the custom columns that **do not** contain data that is already present in Drupal
- [ ] PM reviews content from audit
- [x] Add these fields from the audit to the documentation entity
   - [x] Label: Pushed to
      - data_pushed_to
      - entity reference: 
      - cardinality: unlimited
      - description: Choose the destination(s) where this data is pushed
   - [x] Label: Pulled from
       - data_pulled_from
      - entity reference: Taxonomy external_data
      - cardinality: unlimited
   - [x] link to design source (Figma? and other sources.. like issue history (maybe break this into two fields)
      - label: Design source
      - link_to_design
      - cardinality: unlimited
   - [x] link to cms design system (storybook)
      - label: CMS Design system
      - link_to_cms_design
      - cardinality: unlimited
   - [x] link to VA design system
      - VA Design system
      - link_to_va_design
      - cardinality: unlimited
   - [x] link to KB artiicles (cardinality multiple)
      - label: Related KB articles 
      - type: entity reference to KB article content type
      - cardinality: multiple
   - [x] supporting links  
      - label: Supporting links -This is more of a free form collection of links that are not as structured as the others
      - cardinality unlimited
   - [x] stakeholders / owned by
      - entity reference to product taxonomy 
      - label: Product stakeholder
      - description: What products own or utilize this?
      - cardinality: unlimited
   - [x]  screenshot ?
      - image field
   - [-] supporting modules https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12309#issuecomment-1425916141
   - [-] links to where it appears on FE - Decided the use of this could fall under "other"  since all nodes have a link to their FE counterpart.  In some cases this might be helpful but in most cases, the content may move, so better to stay with the links on the node.
   - [x] links to research findings
   - [ ] 

new feature related to number of editors????

### PR DESCRIPTION
**Pull request by swirtSJW**
## Description

closes #12309

## QA steps / Testing done

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As an admin
1. Go to https://pr12580-t6emjy612llid3a4knyd7qbp1c4kzi9s.ci.cms.va.gov/admin/structure/cm_document/add
   - Design Group
      - [ ] validate Links to Design exist as a field
      - [ ] validate Links to CMS Design System exist as a field
      - [ ] validate Links to research exist as a field
      - [ ] validate Links to VA Design System exist as a field
   - [ ] Validate "Related knowlegebase articles" exist
   - [ ] Validate Stakeholders / Owners exist
   - [ ] Validate Other links field exists
   - [ ] Validate screenshots upload exists.
   -  Data group
      - [ ] Validate "Data pushed to" field exists
      - [ ] Validate "Data pulled from" field exists

- [ ] https://pr12580-t6emjy612llid3a4knyd7qbp1c4kzi9s.ci.cms.va.gov/admin/structure/taxonomy/manage/external_data_source_destination/overview validate three terms exist.
- [x] proof of operation of the deploy hook
image

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [x] Tests have been added if necessary.
- [x] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by swirtSJW** [comment]
@laflannery I think we can forgo the 508 review and apply it as part of 
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12574  Since design may call for widget changing or grouping.

File: docroot/modules/custom/va_gov_db/va_gov_db.deploy.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * This is a NAME.deploy.php file. It contains "deploy" functions. These are
+ * one-time functions that run *after* config is imported during a deployment.
+ * These are a higher level alternative to hook_update_n and
+ * hook_post_update_NAME functions.
+ *
+ * See https://www.drush.org/latest/deploycommand/#authoring-update-functions
+ * for a detailed comparison.
+ */
+
+require_once __DIR__ . '/../../../../scripts/content/script-library.php';
```
Comment (timcosgrove) [inline docroot/modules/custom/va_gov_db/va_gov_db.deploy.php:16]:
```suggestion
require_once $_SERVER['DOCUMENT_ROOT'] . '/../scripts/content/script-library.php';
```
maybe?

**Review by timcosgrove — COMMENTED** [review]
Just small code thing

**Comment by timcosgrove** [comment]
## QA steps / Testing done

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As an admin
1. Go to https://pr12580-t6emjy612llid3a4knyd7qbp1c4kzi9s.ci.cms.va.gov/admin/structure/cm_document/add
   - Design Group
      - [ ] validate Links to Design exist as a field
      - [ ] validate Links to CMS Design System exist as a field
      - [ ] validate Links to research exist as a field
      - [ ] validate Links to VA Design System exist as a field
   - [ ] Validate "Related knowlegebase articles" exist
   - [ ] Validate Stakeholders / Owners exist
   - [ ] Validate Other links field exists
   - [ ] Validate screenshots upload exists.
   -  Data group
      - [ ] Validate "Data pushed to" field exists
      - [ ] Validate "Data pulled from" field exists

- [ ] https://pr12580-t6emjy612llid3a4knyd7qbp1c4kzi9s.ci.cms.va.gov/admin/structure/taxonomy/manage/external_data_source_destination/overview validate three terms exist.
- [x] proof of operation of the deploy hook
image



### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [x] Tests have been added if necessary.
- [x] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.




### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

File: config/sync/core.entity_form_display.cm_document.cm_document.default.yml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    settings:
+      match_operator: CONTAINS
+      match_limit: 10
+      size: 60
+      placeholder: ''
+    third_party_settings:
+      limited_field_widgets:
+        limit_values: '0'
+      change_labels:
+        add_another: 'Add another knowledgebase article link'
```
Comment (timcosgrove) [inline config/sync/core.entity_form_display.cm_document.cm_document.default.yml:159]:
```suggestion
        add_another: 'Add another Knowledge Base article link'
```

File: config/sync/field.field.cm_document.cm_document.field_related_knowledgebase.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    circular_reference_deep: false
+    duplicate_reference: false
+  epp:
+    value: ''
+    on_update: 0
+id: cm_document.cm_document.field_related_knowledgebase
+field_name: field_related_knowledgebase
+entity_type: cm_document
+bundle: cm_document
+label: 'Related knowledgebase articles'
```
Comment (timcosgrove) [inline config/sync/field.field.cm_document.cm_document.field_related_knowledgebase.yml:24]:
```suggestion
label: 'Related Knowledge Base articles'
```

File: config/sync/field.field.cm_document.cm_document.field_related_knowledgebase.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+description: 'List any knowledgebase articles that may be related.'
```
Comment (timcosgrove) [inline config/sync/field.field.cm_document.cm_document.field_related_knowledgebase.yml:25]:
```suggestion
description: 'List any Knowledge Base articles that may be related.'
```

File: config/sync/field.field.cm_document.cm_document.field_links_to_design.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+third_party_settings:
+  epp:
+    value: ''
+    on_update: 0
+id: cm_document.cm_document.field_links_to_design
+field_name: field_links_to_design
+entity_type: cm_document
+bundle: cm_document
+label: 'Links to Design'
+description: 'Add links to design related artifacts (Figma, Github Issue or Epic, relavant Slack thread)'
```
Comment (timcosgrove) [inline config/sync/field.field.cm_document.cm_document.field_links_to_design.yml:20]:
```suggestion
label: 'Links to design'
description: 'Add links to design-related artifacts (Figma, Github Issue or Epic, relevant Slack thread).'
```

File: config/sync/field.field.cm_document.cm_document.field_links_to_research.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+third_party_settings:
+  epp:
+    value: ''
+    on_update: 0
+id: cm_document.cm_document.field_links_to_research
+field_name: field_links_to_research
+entity_type: cm_document
+bundle: cm_document
+label: 'Links to research'
+description: 'Add links to research'
```
Comment (timcosgrove) [inline config/sync/field.field.cm_document.cm_document.field_links_to_research.yml:20]:
```suggestion
label: 'Links to research'
description: 'Add links to research.'
```

File: config/sync/field.field.cm_document.cm_document.field_link_to_va_design_system.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+third_party_settings:
+  epp:
+    value: ''
+    on_update: 0
+id: cm_document.cm_document.field_link_to_va_design_system
+field_name: field_link_to_va_design_system
+entity_type: cm_document
+bundle: cm_document
+label: 'Links to VA Design System (FE)'
+description: 'Links to relevant <a href="https://design.va.gov/">VA Design</a> System documentation'
```
Comment (timcosgrove) [inline config/sync/field.field.cm_document.cm_document.field_link_to_va_design_system.yml:20]:
```suggestion
label: 'Links to VA Design System (FE)'
description: 'Add links to relevant <a target="_blank" href="https://design.va.gov/">VA Design System</a> documentation.'
```

File: config/sync/field.field.cm_document.cm_document.field_other_links.yml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+third_party_settings:
+  epp:
+    value: ''
+    on_update: 0
+id: cm_document.cm_document.field_other_links
+field_name: field_other_links
+entity_type: cm_document
+bundle: cm_document
+label: 'Other links'
+description: 'Links to anything else that did not fit into an existing category. (examples: link to FE sample, links to exceptions, links to related documentation ...)'
```
Comment (timcosgrove) [inline config/sync/field.field.cm_document.cm_document.field_other_links.yml:20]:
```suggestion
label: 'Other links'
description: 'Add links to anything else that did not fit into an existing category - for example, links to FE samples, links to exceptions, links to related documentation.'
```

File: config/sync/field.field.cm_document.cm_document.field_screenshots.yml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    - image
+third_party_settings:
+  epp:
+    value: ''
+    on_update: 0
+id: cm_document.cm_document.field_screenshots
+field_name: field_screenshots
+entity_type: cm_document
+bundle: cm_document
+label: screenshots
```
Comment (timcosgrove) [inline config/sync/field.field.cm_document.cm_document.field_screenshots.yml:19]:
```suggestion
label: Screenshots
```

File: config/sync/field.field.cm_document.cm_document.field_screenshots.yml [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+description: 'Add screenshots for both the Look of the edit form after a design change and the look of the view mode.'
```
Comment (timcosgrove) [inline config/sync/field.field.cm_document.cm_document.field_screenshots.yml:20]:
```suggestion
description: 'Add screenshots for both the look of the edit form after a design change and the look of the view mode.'
```

**Review by timcosgrove — APPROVED** [review]
Approved. I added some spelling/grammar/wording suggestions; feel free to use as you like.

**Comment by swirtSJW** [comment]
Switch to using SERVER var resulted in failure.  
image

Reverting that change.

File: docroot/modules/custom/va_gov_db/va_gov_db.deploy.php [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+ * This is a NAME.deploy.php file. It contains "deploy" functions. These are
+ * one-time functions that run *after* config is imported during a deployment.
+ * These are a higher level alternative to hook_update_n and
+ * hook_post_update_NAME functions.
+ *
+ * See https://www.drush.org/latest/deploycommand/#authoring-update-functions
+ * for a detailed comparison.
+ */
+
+require_once $_SERVER['DOCUMENT_ROOT'] . '/../scripts/content/script-library.php';
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_db/va_gov_db.deploy.php:16]:
```suggestion
require_once __DIR__ . '/../../../../scripts/content/script-library.php';
```

**Comment by swirtSJW** [comment]
image

**Pull request merged**